### ISSUE TITLE
Patch section in a separate file

### ISSUE THREAD

**Issue by wyfo** [rust-lang/cargo#10832]
### Problem

I would like to use `patch` section to do local development, patching a dependency to use a `path`. However I don't want to pollute `Cargo.toml` with local/temporary stuff, because `Cargo.toml` is versioned.

### Proposed Solution

Allow for a new file, named `Cargo-patch.toml` (or whatever name better suited). When the file is present, its content is added to the `patch` section of `Cargo.toml`, overriding items present in both files.

This feature could be turned off (with an environment variable for example) to use *unpatched* `Cargo.toml` with tools like pre-commit.

**Comment by ehuss** [comment]
Patches can also be specified in a [config file](https://doc.rust-lang.org/nightly/cargo/reference/config.html#patch). Does that work for you?

**Comment by wyfo** [comment]
By the way, maybe the documentation part about [Overriding Dependencies](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#overriding-dependencies) is a little bit out of date, because `.cargo/config.toml` is only mentioned for [paths overrides](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#paths-overrides).
(In my case, I wanted to patch git dependencies, so *paths overrides* didn't work)

Should I open a new issue about documentation update?

**Comment by ehuss** [comment]
Nah, I just opened a quick PR at #10836 to include it in the docs. Thanks for the suggestion! I completely missed that we didn't update that chapter.  (Config support was just recently added.)
